### PR TITLE
Set SATA as the default CDROM controller and fix Windows mounting tools iso issue

### LINKED ISSUE #143: [Bug] [Windows] Mount VMware Tools ISO file failed with CD-ROM attach to IDE controller not support hot-add
##### SUMMARY

When run testing on the existing VM, there is no IDE CDROM and default it's SATA CDROM, so add a new IDE CDROM will fail when VM is power on.

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
"ansible [core 2.11.5] ",
        "  config file = /root/ansible-vsphere-gos-validation/ansible.cfg",
        "  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']",
        "  ansible python module location = /usr/lib/python3.9/site-packages/ansible",
        "  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections",
        "  executable location = /usr/bin/ansible",
        "  python version = 3.9.1 (default, Aug 19 2021, 02:58:42) [GCC 10.2.0]",
        "  jinja version = 3.0.1",
        "  libyaml = True"
```

##### CONFIGURATION

```paste below
        "CALLBACKS_ENABLED(/root/ansible-vsphere-gos-validation/ansible.cfg) = ['timer']",
        "DEFAULT_CALLBACK_PLUGIN_PATH(/root/ansible-vsphere-gos-validation/ansible.cfg) = ['/root/ansible-vsphere-gos-validation/plugin']",
        "DISPLAY_SKIPPED_HOSTS(/root/ansible-vsphere-gos-validation/ansible.cfg) = False",
        "RETRY_FILES_ENABLED(/root/ansible-vsphere-gos-validation/ansible.cfg) = False"
```

##### ENVIRONMENT / OS

Windows

##### STEPS TO REPRODUCE

Existing VM with SATA CDROM

##### EXPECTED RESULTS

wintools_complete_install_verify passed

##### ACTUAL RESULTS

wintools_complete_install_verify failed on existing VM

### LINKED ISSUE #270: [New Feature] Change to use SATA CDROM as the default one
##### SUMMARY

Now by default use IDE CDROMs, we did not consider SATA boot disk situation when using SATA CDROM, so this issue is used for tracking such changes.

##### ISSUE TYPE
- Feature Idea

##### ENVIRONMENT / OS

All

### PR DESCRIPTION
**Pull request by Tomorrow9**
Fixes #143 
Fixes #270

### CODE REVIEW

File: common/download_iso_and_transfer.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@
   debug: var=transferred_install_iso
 
 - name: Copy downloaded OS installation ISO file to ESXi host
-  include_tasks: ../../common/transfer_file_remote.yml
+  include_tasks: transfer_file_remote.yml
```
Comment (keirazhang) [inline common/download_iso_and_transfer.yml:36]: I think here can replace with esxi_upload_datastore_file.yml.

Comment (Tomorrow9) [reply common/download_iso_and_transfer.yml]: Yeah, right, changed it to esxi_upload_datastore_file.yml. Thanks.

**Review by keirazhang — DISMISSED** [review]
LGTM. Thanks!

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**